packages/tools/client-debugger/client-debugger-view/src/test/widgets/EmojiButton.tsx

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { TooltipHost } from "@fluentui/react";
+import { Button, useId } from "@fluentui/react-components";
+import React from "react";
+
+import { SharedCell } from "@fluidframework/cell";
+
+/**
+ * {@link EmojiButton} input props.
+ */
+export interface EmojiButtonProps {
+	emojiCell: SharedCell<boolean>;
+}
+
+/**
+ * Simple button that displays either a smily or frouny emoji.
+ * Pressing the button toggles between the two.
+ *
+ * State is shared via the provided `SharedCell`.
+ */
+export function EmojiButton(props: EmojiButtonProps): React.ReactElement {
+	const { emojiCell } = props;
+
+	// undefined => No expression
+	// false => frouny
+	// true => smily
+	const [isSmily, setIsSmily] = React.useState<boolean | undefined>(emojiCell.get());
+
+	React.useEffect(() => {
+		function updateState(): void {
+			setIsSmily(emojiCell.get());
+		}
```

Comment:
nitpick: Since this is a function defined in local scope of the useEffect() I'm not sure if there's any benefit doing this over just calling setIsSmily(emojiCell.get());

Reply:
I think the point of the function is that it acquires an "identity" so we can add it as an event handler and then remove it with `emojiCell.off()`. That wouldn't work with anonymous functions declared in the on()/off() calls, since they would be different objects.

Reply:
Yeah, that's why it's needed.